packages/fuchsia_remote_debug_protocol/lib/src/common/logging.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:core';
+
+/// Determines the level of logging.
+///
+/// Verbosity is increasing from zero (none) to four (fine). The fifth level
+/// (all) logs everything.
+enum LoggingLevel {
+  /// Logs no logs.
+  none,
+
+  /// Logs severe logs at the most.
+  severe,
+
+  /// Logs warning logs at the most.
+  warning,
+
+  /// Logs info logs at the most.
+  info,
+
+  /// Logs fine logs at the most.
+  fine,
+
+  /// Logs everything.
+  all,
+}
+
+/// Synchronously logs a `Log.`
+typedef void LoggingFunction(Log log);
+LoggingFunction _defaultFormatter = (Log log) {
+  print('[${log.levelName}] -- ${log.time}: ${log.message}');
```

Comment:
Curious why we want to omit the `tag` by default.

Reply:
Just figured it'd be nice to have optionally.  I added it in anyway